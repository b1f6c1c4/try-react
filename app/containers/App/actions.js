import {
  TOGGLE_DRAWER,
} from './constants';

export function toggleDrawer() {
  return {
    type: TOGGLE_DRAWER,
  };
}
