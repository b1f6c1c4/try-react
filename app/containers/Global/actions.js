import {
  DEFAULT_ACTION,
  EXTERNAL_REQUEST,
  EXTERNAL_SUCCESS,
  EXTERNAL_FAILURE,
} from './constants';

export function defaultAction() {
  return {
    type: DEFAULT_ACTION,
  };
}


export function externalRequest() {
  return {
    type: EXTERNAL_REQUEST,
  };
}

export function externalSuccess(result) {
  return {
    type: EXTERNAL_SUCCESS,
    result,
  };
}

export function externalFailure(error) {
  return {
    type: EXTERNAL_FAILURE,
    error,
  };
}
