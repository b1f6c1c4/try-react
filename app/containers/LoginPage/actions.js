import {
  SUBMIT_LOGIN_ACTION,
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
} from './constants';

// Actions
export function submitLoginAction() {
  return {
    type: SUBMIT_LOGIN_ACTION,
  };
}

// Sagas
export function loginRequest() {
  return {
    type: LOGIN_REQUEST,
  };
}

export function loginSuccess(result) {
  return {
    type: LOGIN_SUCCESS,
    result,
  };
}

export function loginFailure(error) {
  return {
    type: LOGIN_FAILURE,
    error,
  };
}
