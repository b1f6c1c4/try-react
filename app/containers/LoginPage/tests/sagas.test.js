import { put } from 'redux-saga/effects';

import * as loginPageActions from '../actions';

import {
  handleLoginRequest,
} from '../sagas';

const credential = 'the-credential';

// Sagas
/* eslint-disable redux-saga/yield-effects */
describe('handleLoginRequest Saga', () => {
  let generator;

  beforeEach(() => {
    generator = handleLoginRequest();

    const selectDescriptor = generator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = generator.next(credential).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the loginPageActions.loginSuccess action if it requests the credential successfully', () => {
    const response = [{
      name: 'First',
    }, {
      name: 'Second',
    }];
    const putDescriptor = generator.next(response).value;
    expect(putDescriptor).toEqual(put(loginPageActions.loginSuccess(response)));
  });

  it('should call the loginPageActions.loginFailure action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = generator.throw(response).value;
    expect(putDescriptor).toEqual(put(loginPageActions.loginFailure(response)));
  });
});
