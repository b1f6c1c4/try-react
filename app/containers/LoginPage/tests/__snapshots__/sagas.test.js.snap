// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handleLoginRequest Saga should call the loginPageActions.loginFailure action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`handleLoginRequest Saga should call the loginPageActions.loginFailure action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/login",
      undefined,
      "the-credential",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`handleLoginRequest Saga should dispatch the loginPageActions.loginSuccess action if it requests the credential successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`handleLoginRequest Saga should dispatch the loginPageActions.loginSuccess action if it requests the credential successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [
      "/login",
      undefined,
      "the-credential",
    ],
    "context": null,
    "fn": [Function],
  },
}
`;
